### PR TITLE
[WFCORE-5279] Split Elytron SSL into its own subsystem and layer

### PR DESCRIPTION
**Pull request by jessicarod7**
PoC for elytron-tls-subsystem. Will derive from changes in [ELY-2084 branch](https://github.com/wildfly-security/wildfly-elytron/pull/1729) of `wildfly-elytron`. Continuing from #1.

Related issues:
- https://issues.redhat.com/browse/WFCORE-5279
- https://issues.redhat.com/browse/ELY-2084
- https://issues.redhat.com/browse/EAP7-1564

### CODE REVIEW

File: feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/elytron-tls-dependency/main/module.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  ~ limitations under the License.
+  -->
+<module xmlns="urn:jboss:module:1.9" name="org.wildfly.elytron-tls-dependency">
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
+    <resources>
+        <artifact name="${org.wildfly.extras.elytron-tls:elytron-tls-dependency}"/>
+        <artifact name="${org.wildfly.security:wildfly-elytron-ssl-common"></artifact>
```
Comment (skyllarr) [inline feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/elytron-tls-dependency/main/module.xml]: Just a minor, missing `}` at the end

File: feature-pack/pom.xml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            <groupId>${project.groupId}</groupId>
+            <artifactId>elytron-tls-dependency</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>elytron-tls-subsystem</artifactId>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
```
Comment (skyllarr) [inline feature-pack/pom.xml:146]: @cam-rod  Just a minor, missing
```
<dependency>
            <groupId>org.wildfly.security</groupId>
            <artifactId>wildfly-elytron-ssl-common</artifactId>
        </dependency>
```
dependency here

File: feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        <!-- Typical module dependencies for a subsystem -->
+        <module name="org.jboss.as.controller"/>
+        <module name="org.jboss.as.server"/>
+        <module name="org.jboss.logging"/>
+        <module name="org.jboss.modules"/>
+        <module name="org.jboss.staxmapper"/>
+        <module name="org.jboss.weld.api"/>
+        <module name="org.jboss.weld.core"/>
+        <module name="org.jboss.weld.spi"/>
+        <module name="org.wildfly.security.base"/>
```
Comment (skyllarr) [inline feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml]: Definition of module "org.wildfly.security.base" is currently missing

Comment (skyllarr) [reply feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml]: likely it shoild be  "org.wildfly.security.elytron-base"

File: feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        <!-- TODO: Remove this module once key store and manager objects are implemented -->
+        <module name="org.wildly.extension.elytron"/>
```
Comment (skyllarr) [inline feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml]: @cam-rod Just a minor, s/wildly/wildfly

Comment (skyllarr) [reply feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml]: I could build without running of tests after adding following modules to the dependencies of this module:
```
<module name="org.wildfly.security.elytron-base"/>
        <!-- TODO: Remove this module once key store and manager objects are implemented -->
        <module name="org.wildfly.extension.elytron"/>
        <module name="java.security.sasl"/>
        <module name="org.wildfly.common"/>
```

File: feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml (hunk 3 above)
Comment (jessicarod7) [reply feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml]: thanks!

File: feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml (hunk 4 above)
Comment (jessicarod7) [reply feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/elytron-tls/main/module.xml]: thanks, added

**Comment by jessicarod7** [comment]
Removing this PR since the changes needed to implement the new Service API are rather extensive and undoing the changes directly is unfeasible. Will open a new PR for WFCORE-5279 using the legacy Service API.

This branch and its changes was renamed to [new-service-template](https://github.com/cam-rod/elytron-tls/tree/new-service-template) and might be used later on as method to implement the new API. Relevant files are:

- the [runtime folder](https://github.com/cam-rod/elytron-tls/tree/new-service-template/subsystem/src/main/java/org/wildfly/extension/elytron/tls/subsystem/runtime) and it's classes
- [KeyStoreDefinition](https://github.com/cam-rod/elytron-tls/blob/new-service-template/subsystem/src/main/java/org/wildfly/extension/elytron/tls/subsystem/KeyStoreDefinition.java)
- [KeyStoreService](https://github.com/cam-rod/elytron-tls/blob/new-service-template/subsystem/src/main/java/org/wildfly/extension/elytron/tls/subsystem/KeyStoreService.java)

**Pull request closed without merging**